content/js1/sprints/4/prep/index.md

```diff
@@ -8,4 +8,666 @@ backlog= 'Module-JS1'
+>
+> "Cool, how are you going to check it works?"
+>
+> "We can use tests to check that the function gives the target output as described in the specification"
+>
+> "No problem, how are you going to write a lot of tests efficiently?"
+>
+> "We can make use of a testing framework - this will help us to write test cases quickly. And it will ensure that the tests give us fast, reliable feedback."
+
+### 🎒 Starting a project
+
+Let's imagine we've started a brand new project in a directory called `week-4-example`.
+
+{{<note type="exercise" title="exercise">}}
+
+### 💻 Follow along
+
+Set up a new directory on your local machine called `week-4-example`.
+Make sure you're inside the `cyf` directory you created earlier in the course.
+Change directory into the this place and double check your current working directory.
+
+```bash
+$ pwd
+.../cyf/week-4-example
+```
+
+{{</note>}}
+
+To start, we need to create a file called `package.json` in our project.
+
+> A `package.json` is used to store useful information about our current project, like the name, description, version etc.
+
+We can create a `package.json` file and add the following to it:
+
+package.json
+
+```json
+{
+  "name": "week-4-test-example"
+}
+```
+
+Anyone using this project can now check the `package.json` to get the name of the project.
+We can also add a description:
+
+```json
+{
+  "name": "week-4-test-example",
+  "description": "An example application showing how to write tests using the jest framework"
+}
+```
+
+We can continue adding more information about our project as the project grows.
+For now we can double check we only have a `package.json` in our project:
+
+```bash
+$ ls
+package.json
+```
+
+### 📦 Using packages
+
+When writing software, we need to continually make use of **packages** written by other developers.
+
+A **package** is a set of programs which are grouped together to provide some functionality.
+
+Different programming languages give developers different ways of accessing packages for use in their code.
+
+In JavaScript, we can use **npm**.
+
+> **npm** is a tool used for accessing packages that can be downloaded and used to help developers write applications. The **npm registry** is an online collection of packages that we can download using **npm** commands.
+
+### 🃏 Installing Jest
+
+Our next step will be to figure out how to install the Jest package on our machine.
+
+We can find out more about the Jest framework from the [documentation online](https://jestjs.io/docs/getting-started).
+
+![jest-install](jest-install.png)
+
+In the **Getting started** section of the documentation, Jest gives us the following command:
+
+```bash
+npm install --save-dev jest
+```
+
+- `npm` is an online database of packages, like Jest, that we can fetch and install on to our machines. It also provides us with
+- in the terminal, we can use the `npm` command to install packages on the
+
+`install` - download a package on to our machine.
```

Comment:
This feels like it's missing an introduction where we said something like "Let's break down the individual parts of that command line" (and where `npm` was one of them)